docs/proposals/scheduler/controller.proto

```diff
@@ -60,11 +48,23 @@ message WorkloadStatus {
     }
 
     string name = 1;
-    StatusCode status_code = 2;
+    Status status = 2;
     Resources resource_usage = 3;
-    string message = 4;
+}
+
+message Empty {}
+
+message WorkloadSignal {
+    enum Signal {
+        STOP = 0;
+        KILL = 1;
+    }
+
+    string name = 1;  // unique name identifier
+    Signal signal = 2;
 }
 
 service SchedulingService {
     rpc Schedule(SchedulingRequest) returns (stream WorkloadStatus);
+    rpc Signal (WorkloadSignal) returns (Empty) {}
```

Comment:
A signal is an agent specific implementation, the client and thus the controller do not want to "signal" and instance, they want to stop or destroy it. Thus the scheduler should only offer specific actions on created instances (e.g. `stop` and `kill/destroy`). And then the scheduler would translate those actions into actual signal calls.

Reply:
Thanks for the input! Methods `stop` and `destroy` have been created on the Scheduler.